File: passenger-service/src/main/java/com/example/passenger_service/model/dto/CompleteProfileRequestDto.java
package com.example.passenger_service.model.dto;

import com.example.passenger_service.model.enums.Gender;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.Builder;

@Builder
public record CompleteProfileRequestDto(
        @NotBlank(message = "Имя пользователя должно быть!")
        String name,

        @Pattern(regexp = "^\\+?[0-9]{10,15}$",
                message = "Неверный формат номера телефона!")
        @NotBlank(message = "Номер телефона должен быть!")
        String phoneNumber,

        @NotNull
        Gender gender
) {
}

File: driver-service/src/main/kotlin/com/example/driver_service/model/dto/CompleteProfileRequestDto.kt
package com.example.driver_service.model.dto

import com.example.driver_service.model.enums.Gender
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Pattern

data class CompleteProfileRequestDto(
    @field:NotBlank(message = "Имя обязательно")
    val name: String,

    @field:NotBlank(message = "Номер телефона обязателен")
    @field:Pattern(regexp = "^\\+?[0-9]{10,15}$", message = "Неверный формат номера телефона")
    val phoneNumber: String,

    @field:NotNull(message = "Пол обязателен")
    val gender: Gender
)